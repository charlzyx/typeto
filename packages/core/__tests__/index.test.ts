import { describe, it, expect } from "bun:test";
import {
  removeEmptyComments,
  parseAndRemoveAnnotations,
} from "../src/getNodeExtraInfo";

describe("removeEmptyComments", () => {
  it("应该移除空注释 / should remove empty comments from the input string", () => {
    const input = `
      /**
       * 
       */
      // 
      // another comment
    `;
    const result = removeEmptyComments(input.trim());
    expect(result).toMatchSnapshot();
  });

  it("当没有空注释时应返回原字符串 / should return the original string if no empty comments are present", () => {
    const input = `
      // another comment
    `;
    const result = removeEmptyComments(input.trim());
    expect(result).toMatchSnapshot();
  });

  it("应该移除多行注释中的空行 / should remove empty lines in multi-line comments", () => {
    const input = `
      /**
       * 
       * 这是一段描述
       * 
       */
      // 
      // another comment
    `;
    const result = removeEmptyComments(input.trim());
    expect(result).toMatchSnapshot();
  });

  it("应该移除单行注释 / should remove single line comments", () => {
    const input = `
      // 这是一个注释
    `;
    const result = removeEmptyComments(input.trim());
    expect(result).toMatchSnapshot();
  });

  it("应该处理没有注释的情况 / should handle cases with no comments", () => {
    const input = ``;
    const result = removeEmptyComments(input.trim());
    expect(result).toMatchSnapshot();
  });
});

describe("parseAndRemoveAnnotations", () => {
  it("应该解析注解并从输入字符串中移除它们 / should parse annotations and remove them from the input string", () => {
    const input = "@title(Example) some text @flag";
    const result = parseAndRemoveAnnotations(input);
    expect(result).toMatchSnapshot();
  });

  it("当没有注解时应返回空的解析对象和清理后的字符串 / should return empty parsed object and cleaned string if no annotations are present", () => {
    const input = "some text without annotations";
    const result = parseAndRemoveAnnotations(input);
    expect(result).toMatchSnapshot();
  });
});
