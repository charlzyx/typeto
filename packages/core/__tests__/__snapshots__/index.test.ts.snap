// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`removeEmptyComments 应该移除空注释 / should remove empty comments from the input string 1`] = `"another comment"`;

exports[`removeEmptyComments 当没有空注释时应返回原字符串 / should return the original string if no empty comments are present 1`] = `"another comment"`;

exports[`removeEmptyComments 应该移除多行注释中的空行 / should remove empty lines in multi-line comments 1`] = `
"这是一段描述
another comment"
`;

exports[`removeEmptyComments 应该移除单行注释 / should remove single line comments 1`] = `"这是一个注释"`;

exports[`removeEmptyComments 应该处理没有注释的情况 / should handle cases with no comments 1`] = `""`;

exports[`parseAndRemoveAnnotations 应该解析注解并从输入字符串中移除它们 / should parse annotations and remove them from the input string 1`] = `
{
  "cleaned": "some text",
  "parsed": {
    "flag": true,
    "title": "Example",
  },
}
`;

exports[`parseAndRemoveAnnotations 当没有注解时应返回空的解析对象和清理后的字符串 / should return empty parsed object and cleaned string if no annotations are present 1`] = `
{
  "cleaned": "some text without annotations",
  "parsed": {},
}
`;
